### PR TITLE
make operator more resilient when creating ownership ConfigMap fails

### PR DESCRIPTION
**Pull request by mhenriks**
Signed-off-by: Michael Henriksen <mhenriks@redhat.com>

**What this PR does / why we need it**:

Operator should not "fail" a CDI CR when it cannot create the ownership ConfigMap.  Addresses: https://bugzilla.redhat.com/show_bug.cgi?id=1751193

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

https://bugzilla.redhat.com/show_bug.cgi?id=1751193

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/operator/controller/controller.go [hunk 1]
```diff
@@ -230,23 +223,24 @@ func shouldTakeUpdatePath(logger logr.Logger, targetVersion, currentVersion stri
...
+
 	if err := r.crInit(cr); err != nil {
-		MarkCrFailed(cr, "CrInitError", "Unable to Initialize CR")
 		return reconcile.Result{}, err
 	}
 
 	logger.Info("Successfully entered Deploying state")
 
-	return r.reconcileUpdate(logger, cr)
+	// since the CR was updated above, reconcile will be imediately triggered again
```
Comment (awels) [inline pkg/operator/controller/controller.go]: Typo: immediately

Comment (mhenriks) [reply pkg/operator/controller/controller.go]: unit tests broke so had to go back to calling reconcileUpdate here

**Comment by awels** [comment]
Did you rebase on the latest master? The failure is something I fixed yesterday.

**Comment by mhenriks** [comment]
@awels this branch is current with master

**Comment by awels** [comment]
Yeah, it got past the issue I fixed, now failing on something else.

**Pull request merged**